File: src/main/java/com/vaadin/demo/component/icons/LumoIcons.java
package com.vaadin.demo.component.icons;

import com.vaadin.demo.flow.routing.Route;
import com.vaadin.flow.component.button.Button;
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.component.icon.Icon;
import com.vaadin.flow.component.orderedlayout.HorizontalLayout;
import com.vaadin.flow.theme.lumo.LumoIcon;
import com.vaadin.demo.DemoExporter; // hidden-source-line

@Route("lumo-icons")
public class LumoIcons extends Div {

  public LumoIcons() {
    HorizontalLayout layout = new HorizontalLayout();
    layout.setSpacing(true);
    layout.addClassName("items-center");

    // tag::snippet[]
    Icon phoneIcon = LumoIcon.PHOTO.create();
    Icon calendarIcon = LumoIcon.CALENDAR.create();
    Icon alarmIcon = LumoIcon.CLOCK.create();
    Button button = new Button(LumoIcon.BELL.create());

    layout.add(phoneIcon, calendarIcon, alarmIcon, button);
    // end::snippet[]

    add(layout);
  }

  public static class Exporter extends DemoExporter<LumoIcons> { // hidden-source-line
  }
}

File: frontend/demo/component/icons/lumo-icons.ts
import 'Frontend/demo/init'; // hidden-source-line

import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import { applyTheme } from 'Frontend/generated/theme';
import '@vaadin/horizontal-layout';
import '@vaadin/icon';
import '@vaadin/vaadin-lumo-styles/vaadin-iconset.js';

@customElement('lumo-icons')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    // Apply custom theme (only supported if your app uses one)
    applyTheme(root);
    return root;
  }

  protected override render() {
    return html`
      <vaadin-horizontal-layout theme="spacing" class="items-center">
        <!-- tag::snippet[] -->
        <vaadin-icon icon="lumo:photo"></vaadin-icon>
        <vaadin-icon icon="lumo:calendar"></vaadin-icon>
        <vaadin-icon icon="lumo:clock"></vaadin-icon>
        <vaadin-button theme="icon">
          <vaadin-icon icon="lumo:bell"></vaadin-icon>
        </vaadin-button>
        <!-- end::snippet[] -->
      </vaadin-horizontal-layout>
    `;
  }
}
